partner_compassion/models/partner_compassion.py

```diff
@@ -364,16 +390,7 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
-        email = vals.get("email")
-        if email:
-            vals["email"] = email.strip()
-            # Push email to non-internal users
-            user_ids = self.mapped("user_ids").filtered("share").ids
-            self.env.cr.execute("""
-                UPDATE res_users
-                SET login=%s
-                WHERE id=ANY(%s)
-            """, [vals["email"], user_ids])
+        self = self.sudo()
```

Comment:
I don't think using sudo in the write method is a good idea, it could lead to many security flaws. I would rather add the missing security rules if some are needed.
```suggestion
```